### ISSUE TITLE
Workstation 16.2.0 & 16.2.1 cannot use paste.

### ISSUE THREAD

**Issue by routeToWeb** [vmware/open-vm-tools#555]
Opening an issue on here as advised on this thread: https://communities.vmware.com/t5/VMware-Workstation-Pro/Workstation-16-2-0-cannot-use-paste/m-p/2878291#M172522
`ps aux | grep -i vmtools
root 699 0.2 0.4 273008 4780 ? Ssl 09:53 0:00 /usr/bin/vmtoolsd
vagrant 3779 0.0 0.0 112816 996 tty1 S+ 09:55 0:00 grep --color=auto -i vmtools`

open-vm-tools & open-vm-tools-desktop have been installed and started
Please let me know what other information is need to move this forward

**Comment by johnwvmw** [comment]
Nothing on the community post confirms that open-vm-tools-desktop has been installed.  If it is, then the system was not rebooted or the desktop session was not restarted (logoff & login again).

Let's start from the beginning.  Please provide all the requested information so that VMware engineer's can reproduce the issue(s) in house if needed.

Linux release being used:  ??

What graphical desktop is being used" Gnome, KDE, XFCE, ...?

open-vm-tools* packages installed on the system:
  - for RPM based packages:
        **rpm -qa | grep open-vm-tools**

  - for Ubuntu or Debian Linux:
       **apt list --installed | grep open-vm-tools**

Once that both open-vm-tools and open-vm-tools-desktop packages are installed, logoff the desktop session.   The latter package is needed to coordinate graphical/mouse events between the host and Linux guest and it must be installed before beginning a graphical desktop session.

Next provide the output of

    **ps -aef | grep vmtoolsd**

There should be two vmtools processes running.
  - "vmtoolsd"  - the system vmtoolsd service process
  - "vmtoolsd -n vmusr"  - the graphical desktop user vmtoolsd process

What we are expecting to see is the desktop user process such as contributor "joshiga" indicated in the community posting.

    test-vm 1884 0.2 1.2 644340 49132 ? Sl 14:25 0:02 /usr/bin/vmtoolsd -n vmusr --blockFd 3

The presence of "--blockFd 3" would indicate that you are using an Xorg/X11 graphical client and that the vmtoolsd user process is connected to the graphical session.

The presence of "--uinputFd=4" is an indication that you are using a "Wayland" graphical client.    If that is the case, let's first establish if the Xorg/X11 client will work properly.

To switch to an Xorg/X11 session, logoff the desktop.   When logging back in, adjust the session settings to select Xorg or X11 before hitting return on the password screen.  Once the "--inputFd=3" alone is present, test the DnD and copy/paste functionality.

Please provide all the information requested.  It will help our engineers address the issues and improve the support for the Wayland graphical client if that is where the problem may lie.

**Comment by routeToWeb** [comment]
Thanks for the reply.
CentOS 7
`Kernel 3.10.0-1160.15.2.el7.x86_64 on an x86_64`
I'm not using a desktop. The system is just used for building websites so everything is command line. I keep a library of commands to paste into the system

```
rpm -qa | grep open-vm-tools
open-vm-tools-11.0.5-3.el7_9.3.x86_64
open-vm-tools-desktop-11.0.5-3.el7_9.3.x86_64
```

```
ps -aef | grep vmtoolsd
root 700 1 0 09:40 ? 00:00:00 /usr/bin/vmtoolsd
vagrant 3759 2436 0 09:46 tty1 00:00:00 grep --color=auto vmtoolsd
```

This may or may not be relevant: I have to start Apache as `sudo httpd -k start` `sudo systemctl start httpd` errors so is there an issue with the systemctl command which is then having an effect on what we are trying to achieve above?

**Comment by GJoshi27** [comment]
Thanks for sharing the details. We have filed an internal bug and are tracking this issue.  Apparently the clipboard security fix which we did in WS 16.2.0 (and acknowledged in Release notes of 16.2.0) is too aggressive and have introduced the regression.

**Comment by johnwvmw** [comment]
It has been suggested by other users that reverting to Workstation 16.1.2 will obviously not encounter this problem.

Consider, as a workaround, using an SSH or PuTTy session to run the commands on your CentOS 7 VM.   The copy of commands (text) on the host can be pasted in the SSH or PuTTY terminal window.   That copy/paste is entirely within the desktop of the host.

**Comment by routeToWeb** [comment]
Good point johnwvmw. It's also possible to run shared shell scripts that can be create in Notepad++ or similar

**Comment by GJoshi27** [comment]
WS 16.2.2 Windows Update should fix this problem partially. For the Usecase of Running/Accessing Local VMs 
this issue is fixed and Paste Menu shouldn't be greyed out if dndCP is enabled in Advanced ->Isolation settings of the VM.
 Users who are accessing Remote VMs, this issue still persists.

**Comment by jamesonhacking** [comment]
What about users like myself on a Linux host?

**Comment by GJoshi27** [comment]
This will be fixed in next workstation release. 
Both remaining issues remote vm access and Linux Host related.

**Comment by sixro** [comment]
Hi guys, I don't want to create noise, but I have to share this.
I installed a previous version of vmware (16.1.2) as suggested here and elsewhere, but the issue was always there.
I was trying to have a Guest Debian 11.x on a Host Windows 10.

Again I had running (`ps`) **only and only** 1 process (`vmtoolsd`), but not the other one.
I tried to remove and reinstall `open-vm-tools` and `open-vm-tools-desktop` many times (without restart in the middle, then with restart in the middle) but nothing.

After few minutes I found [this guide](https://wiki.archlinux.org/title/VMware/Install_Arch_Linux_as_a_guest#Drag_and_drop,_copy/paste) on ArchLinux.
Unfortunately I was not able to install `gtkmm3` (I don't have Extra repo configured at the moment), so I skipped it.
I checked whether `vmware-user.desktop` was under autostart and it was there.
BUT, I decided to try anyway to the `vmware-user` command in my `.xinitrc` before the execution of my _window manager_ (`fvwm`) and IT WORKED. I haven't done nothing more... after the restart of (only) x11, I have been able to copy text from Host to Guest (middle mouse) and looking at `ps` eventually I found:
```
root      465       1  0 22:10 ?        00:00:02 /usr/bin/vmtoolsd
myuser    976       1  0 22:19 tty1     00:00:01 /usr/bin/vmtoolsd -n vmusr --blockFd 3
```
